File: apps/mobile/android/app/src/main/java/com/visvine/mobile/data/model/Event.kt
package com.visvine.mobile.data.model

import kotlinx.serialization.Serializable

/** Community event — mirrors src/types/index.ts `Event`. */
@Serializable
data class Event(
    val id: String,
    val communityId: String = "",
    val title: String,
    val description: String? = null,
    val startAt: String,
    val endAt: String? = null,
    val timezone: String? = null,
    val location: EventLocation? = null,
    val hosts: List<String> = emptyList(),
    val capacity: Int? = null,
    val visibility: String = "community",
    val analytics: EventAnalytics = EventAnalytics(),
)

@Serializable
data class EventLocation(
    val label: String = "",
    val address: String? = null,
    val lat: Double? = null,
    val lon: Double? = null,
)

@Serializable
data class EventAnalytics(
    val views: Int = 0,
    val rsvpCount: Int = 0,
    val checkinCount: Int = 0,
)

@Serializable
data class EventsResponse(
    val events: List<Event> = emptyList(),
)

File: apps/mobile/ios/Visvine/Models/Event.swift
import Foundation

/// Community event — mirrors src/types/index.ts `Event`.
struct Event: Codable, Identifiable {
    let id: String
    var communityId: String?
    let title: String
    var description: String?
    let startAt: String
    var endAt: String?
    var timezone: String?
    var location: EventLocation?
    var hosts: [String]?
    var capacity: Int?
    var visibility: String?
    var analytics: EventAnalytics?
}

struct EventLocation: Codable {
    var label: String
    var address: String?
    var lat: Double?
    var lon: Double?
}

struct EventAnalytics: Codable {
    var views: Int
    var rsvpCount: Int
    var checkinCount: Int
}

struct EventsResponse: Codable {
    let events: [Event]
}
